### ISSUE TITLE
[Req]: ilcorsaronero

### ISSUE THREAD

**Issue by Jorman** [Jackett/Jackett#986]
Hi, is possible to add this italian traker to jacket?

**Indexer Name**
IlCorsaroNero

**Type**
Movies & TV shows in HD/SD/Screener..., Audio, Games, Others

**Indexer URL**
http://ilcorsaronero.info/

**Indexer Language**
Italian

**Invite**
Non needed

**Note**
None

Tnx, Jo

**Comment by ShahadatRD** [comment]
what is the procedure of getting in that tracker ? @Jorman as far as i know @kaso17  must have access to that tracker. if that tracker is invite only then send a invite to this email kaso1717@gmail.com

**Comment by armani316** [comment]
@Jorman Correct me if I'm wrong, but I believe this is a public tracker that does not require an invite? Either way, it would be great if you could explain the tracker's category (i.e. Movies, TV, Audio, etc.)

Feel free to follow the template I used in #987

**Comment by Jorman** [comment]
Yep, is a public tracker, if help this is an addon for couchpotato that works with ilcorsaronero
https://github.com/bateman/corsaronero-cp-provider
Actually there's also one for tntvillage.

So for example if you search: test search
I got this
http://ilcorsaronero.info/argh.php?search=test+search
But If I apply some filter for dvd I've
http://ilcorsaronero.info/torrent-ita/20/test%20search.html
for bdrip I've
http://ilcorsaronero.info/torrent-ita/1/test%20search.html
for tv serie I've
http://ilcorsaronero.info/torrent-ita/15/test%20search.html

So there's a kind of category on it. I don't have so many skill on programming, but I can help to make some test or help to translate or explain some site section.

Let me know

**Comment by armani316** [comment]
@Jorman Yes, it has multiple search categories, which is normal since that's the case with almost every tracker, public or private. I have added the 'New tracker' label to this issue.

**Comment by Jorman** [comment]
Yep, tell me if you need info

**Comment by bonny1992** [comment]
I am trying to write the definition myself, but I'm encountering a little problem: how does jackett handle input tags and post requests? Because ilcorsaronero doesn't provide directly the magnet or the torrent link, but instead it "compiles" it on the go.
@armani316

**Comment by armani316** [comment]
@bonny1992 I'll direct your question to @kaso17.

**Comment by bonny1992** [comment]
Oh thanks! @armani316

**Comment by bonny1992** [comment]
@kaso17 
Ok, I tried something but no success for now.
Ilcorsaronero may have three methods to get the torrent (file or magnet):

- Clicking on the "Download" icon in the search list (triggering the post request, which starts the download)

- Sending actually the post request (have not tried sending a custom request by postman or similar software, tho)

- Fetching the "Details" page of the single torrent and then getting the magnet from there

What can I do? Is there any similar tracker that does some of those things?

Also I can't find any documentation on the variables I can use on the yml file, without having to see near every file of the project, can you direct me in the right direction?

**Comment by Jorman** [comment]
Hi, I don't know if this can help, but I found an external implementation for ilcorsaronero and couchpotato
https://github.com/bateman/corsaronero-cp-provider

**Comment by bonny1992** [comment]
@Jorman 
I already knew about that provider (which kinda works on CouchPotato), and that uses the "third" method I listed.

`def getMagnetLink(self, url):
		data = self.getHTMLData(url)
		html = BeautifulSoup(data)
		magnet = html.find('a', attrs={'class': 'forbtn'})['href']
return magnet`

**Comment by Jorman** [comment]
Ok, tnx

**Comment by kaso17** [comment]
Have a look at the hdsky definition, it's using post requests for downloads.
isohunt is an example for indirect downloads (set the download link in the search block to the details page. Then Jackett will use the selector in the toplevel download block to get the actual DL link.

The available variables depend on the context.
The config variables (usernamepassword, etc) are always available.
In the search query have a look at these lines: https://github.com/Jackett/Jackett/blob/master/src/Jackett/Indexers/CardigannIndexer.cs#L1055-L1077
During row parsing you can access already set fields/variables using ".Result.$fielname".
In the "download/before" section additional arguments of the download uri are available. Let me know what you need.

**Comment by bonny1992** [comment]
@kaso17 
First of all, thank you for your reply.
It seems that ilcorsaronero uses a weird method to handle torrent downloads, so it's not so practical to implement, it will be maybe better to fetch the details page of the single torrent and then get the href attribute of the magnet link.
What can I do?

Can I contact you by your email? It'd be easier to communicate.
(Also I'm planning to implement icv-crew and some other trackers, too)

**Comment by kaso17** [comment]
with magnet links indirect download doesn't work as most clients don't support http to magnet link redirects. But to me it looks like the "value" attribute of the download button (input tag) contains the magnet hash. Just build the magnet link form it using prepend/append filters. Have a look at the torrenz2 definition for an example.

**Comment by bonny1992** [comment]
@kaso17 
Ok, I understood... Is there any filter to uppercase the string?

Also, I have some issues with the date parsing, can you help me?
The format is **dd.mm.yy**, how can I put it in the yml file to get it parsed correctly?

**Comment by kaso17** [comment]
Time format needs to be in go lang format.
http://fuckinggodateformat.com/ is highly recommended.
For `dd.mm.yy` the format string would be `02.01.06`

**Comment by dpmdiego** [comment]
Does anyone know how I can get an invitation to the arabafenice or shareisland?
If anyone can send me an invitation to the Arabafenice I would be very grateful.
If anyone is interested, I can send an invitation to TB, it is a Brazilian community. Thank you!

**Comment by dpmdiego** [comment]
Does anyone know how I can get an invitation to the arabafenice or shareisland?
If anyone can send me an invitation to the Arabafenice I would be very grateful.
If anyone is interested, I can send an invitation to TR, it is a Brazilian community. Thank you!

**Comment by kaso17** [comment]
@dpmdiego please stop asking for invites/offering invite trades. You may only ask for an invite if you're planning on adding a tracker/fixing a bug.

**Comment by bonny1992** [comment]
@kaso17 
Nothing, I seem to cannot understand how to make the pull request (still quite new to github, sorry).

If you want, here's the yml file with the definition: http://pastebin.com/EBBSVFt4

**Comment by kaso17** [comment]
Thank you.
Added it via 8163c28c1c57aa9554b157e3519528c5dfc1a72a
Added a urldecode filter for the title and made the itorrents links configurable

**Comment by kaso17** [comment]
I'm also getting less search results when using
http://ilcorsaronero.info/adv/bdrip.html (use the the definition)
compared to
http://ilcorsaronero.info/argh.php?search=bdrip
What's the difference between the two searches?

**Comment by bonny1992** [comment]
That's strange.
With 
http://ilcorsaronero.info/adv/bdrip.html
I'm getting 50 results, instead with
http://ilcorsaronero.info/argh.php?search=bdrip
I'm getting 40 results, which is the max for "normal" search (the first one is the advanced search).

Also, I have made another definition, for "passionetorrent" tracker (Italian private tracker), what can I do? Should I open a new issue to give the file to you like I did before or can I paste it here without opening a new issue?

**Comment by kaso17** [comment]
My concern with the searches is that the advanced search doesn't sem to include the most recent releases. Currently the advanced search is showing one more torrent at the top of the list than the normal search for me.

With a more common keyword it's more visible:
http://ilcorsaronero.info/adv/ita.html vs http://ilcorsaronero.info/argh.php?search=ita
currently the advanced search is 24 releases (~1 day) behind the normal search.
Maybe you can contact the site admins to clarify this behaviour?

Please open another issue for passionetorrent

**Comment by bonny1992** [comment]
Oh I now notice what are you saying.
I contacted the admins through the email address they put it in http://ilcorsaronero.info/contatti page, but I'm not so sure they'll reply (soon or at all).

If unsure we can always switch from the advanced search to the normal one in just a moment.

I'll open that right now, then.

**Comment by RoloSoze** [comment]
I'm closing this issue since Il Corsaro Nero has already been added,

**Comment by Jorman** [comment]
Hi, tnx for this. I've one little question about categories, on jacket side, with the plugin I've:

> 100005	Anime
> 100001	BDRiP
> 100020	Full DVD
> 100019	Screener
> 100015	TV Series
> 2030	Movies/SD
> 2040	Movies/HD
> 5000	TV
> 5070	TV/Anime

On sonar, with advanced settings, on categories I see 5030,5040. My question is, I've to put 100015 instead 5030 or I can leave the default one? The categories function is not clear for me. 

Tnx in advance, Cry

**Comment by kaso17** [comment]
The IDs >= 100000 are the tracker specific ones. 
You can see the mapping to the generic ones (<100000) here:
https://github.com/Jackett/Jackett/blob/master/src/Jackett/Definitions/ilcorsaronero.yml#L11-L16
If you set it 5000 it will include the anime category too. If you set it to 100015 it's the "TV Series" category only.

**Comment by Jorman** [comment]
Tnx for your reply, but I don't get it. Maybe I ask the wrong question.

I mean:
On Jackett side this addon have this categories
> 100005	Anime
> 100001	BDRiP
> 100020	Full DVD
> 100019	Screener
> 100015	TV Series
> 2030	Movies/SD
> 2040	Movies/HD
> 5000	TV
> 5070	TV/Anime

But on Sonar side, when I add the ability to search on ilcorsaronero, Sonar, by default, have 5030,5040 categories. So my question is, if I leave 5030, 5040 on Sonar, can be ok? If I got it right, on Sonar, If I set 5000, Sonar will search all over 5000? If yes, way there's 2 categories? 5030 and 5040

**Comment by kaso17** [comment]
5030 (TV/SD) and 5040 (TV/HD) are the default values, it won't work with these.
Use "100015", or "100015, 100005" (if you intend to search for anime).

**Comment by Jorman** [comment]
Ok, so, I only add the categories that this particular site use.

Tnx again

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Error 504 - Gateway time-out

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - 500 - Internal Server Error

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
> https://ilcorsaronero.link/ - 500 - Internal Server Error

**Comment by Jorman** [comment]
back up

**Comment by ilike2burnthing** [comment]
> https://ilcorsaronero.link/ - Error 504 - Gateway time-out

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
> https://ilcorsaronero.link/ - 500 - Internal Server Error

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Error 502 - Bad gateway

**Comment by ilike2burnthing** [comment]
```
Ieri Ilcorsaronero è stato hackerato.
Stiamo lavorando per recuperare i dati.
Se hai utilizzato la stessa passaparola sul nostro sito e qualsiasi altro ti preghiamo di aggiornarla.
Tra 48 ore ci sarà un altro aggiornamento.
Grazie
Scusate per disagio
```
```
Yesterday Ilcorsaronero was hacked.
We are working to recover the data.
If you have used the same word of mouth on our site and any other please update it.
In 48 hours there will be another update.
Thank you
Sorry for inconvenience
```

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Error 522 - Connection timed out

**Comment by gipasoft** [comment]
This link: https://ilcorsaronero-it.cz/ seems to work but in jackett I see this message: 
```
An error occurred while testing this indexer
Found no results while trying to browse this tracker
```

In the log: CardigannIndexer (ilcorsaronero): Relogin required

**Comment by ilike2burnthing** [comment]
Presumably a poor clone

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Error 504 - Gateway time-out

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ returning
```
500 - Internal Server Error
```

**Comment by garfield69** [comment]
today its returning **Error 502 Bad gateway**

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - ERR_CANT_OPEN_DB - Too many connections
https://ilcorsaronero.link/recenti (or any link) - ERR_CANT_OPEN_DB - Access denied

**Comment by ilike2burnthing** [comment]
back up

**Comment by shiner66** [comment]
It's down again, new link seems to be ```https://www.ilcorsaronero.in/```

**Comment by ilike2burnthing** [comment]
I'm not sure that's a new domain for the site, as the homepage doesn't have the usual message from the admins, and the last upload looks to be from over 2wks ago.

https://ilcorsaronero.link/ - 500 Internal Server Error
https://ilcorsaronero.fun/ - Error 504 - Gateway time-out
https://ilcorsaronero.pro/ - Error 522 - Connection timed out

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Error 504 - Gateway time-out

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
*.lin, *.pro, *.fun and .nocensor.sbs all return
```
[modifiche al sito, rimanette in contatto con noi tramite la toolbar](http://ilcorsaronero.ourtoolbar.com/)ERR_CANT_OPEN_DB my_ilcorsaronero - Access denied for user 'www-data'@'localhost' (using password: NO)
```
when doing keywordless search `/browse/0`

https://ilcorsaronero.torrentbay.to/browse/0 works
but fails for keyword searching `https://ilcorsaronero.torrentbay.to/argh.php?search=occhiali+neri` with the same error

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/

> ERR_CANT_OPEN_DB my_ilcorsaronero - Access denied for user 'www-data'@'localhost' (using password: NO)

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Error 504 - Gateway time-out

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
returning ERR_CANT_OPEN_DB my_ilcorsaronero - Too many connections

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
all sitelinks returning Error 504 Gateway time-out

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ returns Origin is unreachable Error code 523

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
all domains: cloudflare Gateway time-out Error code 504

**Comment by ilike2burnthing** [comment]
back up

**Comment by ilike2burnthing** [comment]
https://ilcorsaronero.link/ - Connection timed out - Error code 522

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/issues/986#issuecomment-1483683334

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ ERR_CONNECTION_TIMED_OUT
https://ilcorsaronero.fun/ domain expired
https://ilcorsaronero.pro/ domain expired
https://ilcorsaronero.torrentbay.to/ redirects to https://ilcorsaronero.torrentbay.net/ no response data
https://ilcorsaronero.mrunblock.guru/ redirects to https://ilcorsaronero.mrunblock.life/  cloudflare Gateway time-out Error code 504

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ ERR_CONNECTION_TIMED_OUT

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ cloudflare Gateway time-out Error code 504
as are the other proxies

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ ERR_CONNECTION_TIMED_OUT

**Comment by AARON8888** [comment]
Now site is working.

**Comment by garfield69** [comment]
I'm still getting 504, and all uptime checkpoints are timing out too.

**Comment by AARON8888** [comment]
Sorry @garfield69. You´re right. I made a mistake. I was thinking on other site (Il Corsaro Blu --> https://ilcorsaroblu.org/).

**Comment by garfield69** [comment]
ilcorsaroblu is working too.
Listen, for public and semi-private sites you do not need to report outages, we monitor all of them daily and update tickets accordingly.
the only indexers that we really need status updates on are the private sites, as there are many that we do not have access to, so monitoring by us is sporadic at best.

**Comment by Edgiest05** [comment]
Back up at www.ilcorsaronero.info

**Comment by garfield69** [comment]
Thanks. I'll update the indexer.

**Comment by garfield69** [comment]
seems they have restored from a backup whose latest releases are from 17.04.2015 ?

**Comment by ilike2burnthing** [comment]
Yea, and I can't get the search to work.

**Comment by garfield69** [comment]
I think we should wait a few days and see if things improve. there is not point trying to update an indexer if the web site is not fully functional.

**Comment by Edgiest05** [comment]
Sorry, I had only checked the browse/0/ endpoint to see if the actual site was up and took functionality for granted. I will check if things improve in the future too.

**Comment by Edgiest05** [comment]
Agreed, indexer seems to be abandoned. Thanks for your time on the matter.

**Comment by garfield69** [comment]
v0.21.1802

**Comment by appsettings** [comment]
Seems back online, with https://ilcorsaronero.link

**Comment by shiner66** [comment]
Search is still not working, results are always empty

**Comment by garfield69** [comment]
v0.21.1855

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/issues/986#issuecomment-1902240750

**Comment by ilike2burnthing** [comment]
back up

**Comment by AARON8888** [comment]
It seems to be down again.

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/issues/986#issuecomment-1902240750

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ cloudflare Connection timed out Error code 522

**Comment by garfield69** [comment]
v0.22.685

**Comment by garfield69** [comment]
v0.22.800

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ cloudflare Bad gateway Error code 502

**Comment by garfield69** [comment]
online and on a new engine. indexer will need updating.

**Comment by garfield69** [comment]
https://ilcorsaronero.link/latest > cloudflare Gateway time-out Error code 504

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/issues/986#issuecomment-2682467225

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ - Connection timed out Error code 522
https://ilcorsaronero.torrentbay.st/ > empty response

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.torrentbay.st/ > empty response
https://ilcorsaronero.link/ - online

**Comment by ilike2burnthing** [comment]
back up

**Comment by garfield69** [comment]
https://ilcorsaronero.torrentbay.st/ > empty response

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.torrentbay.st/ > cloudflare Connection timed out Error code 522

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/issues/986#issuecomment-3885308910
https://ilcorsaronero.link/ > online

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
https://ilcorsaronero.link/ > Http timeout
https://ilcorsaronero.torrentbay.st/ > empty response

**Comment by ilike2burnthing** [comment]
[v0.24.1822](https://github.com/Jackett/Jackett/releases/v0.24.1822)

**Comment by ilike2burnthing** [comment]
[v0.24.1827](https://github.com/Jackett/Jackett/releases/v0.24.1827)
